boards/arm/xiao_ble/doc/index.rst

```diff
@@ -82,18 +82,36 @@ LED
 Programming and Debugging
 *************************
 
-The XIAO BLE ships with a bootloader. However, this guide doesn't describe how
-to use it and instead an External Debug Probe is used in order to program the
-board. If you have one, you can also use an external :ref:`debug probe
-<debug-probes>` to flash and debug Zephyr applications, but you need to solder
-an SWD header onto the back side of the board.
+The XIAO BLE ships with the `Adafruit nRF52 Bootloader`_ which supports flashing
+using `UF2`_. Doing so allows easy flashing of new images, but does not support
+debugging the device.
+
+UF2 Flashing
+============
+
+To enter the bootloader, connect the USB port of the XIAO BLE to your host, and
+double tap the reset botton to the left of the USB connector. A mass storage
+device named `XIAO BLE` should appear on the host. Using the command line, or
+your file manager copy the `zephyr/zephyr.uf2` file from your build to the base
+of the `XIAO BLE` mass storage device. The XIAO BLE will automatically reset
+and launch the newly flashed application.
+
+
+External Debugger
+=================
+
+In order to support debugging the device, instead of using the bootloader, you
+can use an External Debug Probe to program the board. If you have one, you can
+also use an external :ref:`debug probe <debug-probes>` to flash and debug
+Zephyr applications, but you need to solder an SWD header onto the back side of
+the board.
```

Comment:
Do you wanna mention the expansion board here? https://wiki.seeedstudio.com/Seeeduino-XIAO-Expansion-Board/

That's what i'm using. it's got pogo pins that match the bottom 4 pins.

Reply:
I don't have one, but that might be a good idea. @mniestroj thoughts? I can add here, or you can add yourself.

Reply:
Yeah, it is a good idea to mention expansion board. Personally for some reason I thought that the expansion board is not compatible with XIAO BLE series.

Reply:
It says on the page that the expansion board is not supported, but there are no reasons given. I asked in their discord, and i couldn't get an answer there either. They do say it will be supported, but also no timeline or mention of the mechanism.

However, I'm able to use the onboard display with an arduino demo sketch and flash the device with jlink/nrfjprog via the pogo pins.